### PR TITLE
Friend notes plugin: Add documentation and increase code readability

### PR DESCRIPTION
**Pull request by SoyChai**
Did some light maintenance as I was documenting this class. See the commit diffs one-by-one.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java [hunk 1]
```diff
@@ -49,6 +49,9 @@ private FriendNoteOverlay(Client client, FriendNotesPlugin plugin, TooltipManage
 		this.tooltipManager = tooltipManager;
 	}
 
+	/**
+	 * Add a friend note tooltip to a hovered friend list entry, if note exists.
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java]: I don't think adding comments to overriden methods is very useful. I would just make this inline comment inside the method

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java [hunk 2]
```diff
@@ -58,14 +61,12 @@ public Dimension render(Graphics2D graphics)
 		}
 
 		final HoveredFriend hovered = plugin.getHoveredFriend();
-		if (hovered == null)
+		if (hovered != null)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java]: Why you flipped this?

Comment (SoyChai) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java]: See the split diff - the method returns null either way so this is easier to follow.

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java (hunk 1 above)
Comment (SoyChai) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java]: Isn't it common for overridden methods to have their own, more specific jdocs? I see this in Java's own standard library all the time.

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java]: Maybe if anyone will ever reference the method directly. But this one is used only by overlay system in generic way, so it should not be here.

Comment (SoyChai) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java]: Alright, I moved those jdocs onto their relevant lines.

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java [hunk 3]
```diff
@@ -139,8 +154,10 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		// look for "Message" on friends list
 		if (groupId == WidgetInfo.FRIENDS_LIST.getGroupId() && event.getOption().equals("Message"))
 		{
-			setCurrentFriend(event.getTarget());
+			final String sanitizedTarget = Text.removeTags(event.getTarget());
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java]: The Text.removeTags is mostly not necessary here, the friend list is not colored.

Comment (SoyChai) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java]: Okay. Do you think it's safe to skip sanitizing the menuTarget in [onMenuOptionClicked()](https://github.com/runelite/runelite/pull/2714/files/11debd38181441f9851ea6310f5e2b9b2c37a11f#diff-39284c20a78808105c9beb0137d2adb0R188) as well?

Comment (Rheon-D) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java]: Nope, that one has color codes in it 😜

Comment (SoyChai) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java]: Done.

**Comment by SoyChai** [comment]
Rebased on post-deob history.

**Pull request merged**